components/cards/ActionForm.tsx

```diff
@@ -201,42 +206,93 @@ export function ActionForm(props: ActionFormProps) {
       content["url"] = parcelWebContentURI;
     }
 
-    if (parcelId && !basicProfileStreamManager) {
-      const assetId = new AssetId({
-        chainId: `eip155:${NETWORK_ID}`,
-        assetName: {
-          namespace: "erc721",
-          reference: licenseAddress.toLowerCase(),
-        },
-        tokenId: parcelId.toString(),
+    const assetId = new AssetId({
+      chainId: `eip155:${NETWORK_ID}`,
+      assetName: {
+        namespace: "erc721",
+        reference: licenseAddress.toLowerCase(),
+      },
+      tokenId: licenseId
+        ? licenseId.toString()
+        : new BN(selectedParcelId.slice(2), "hex").toString(10),
+    });
+
+    if (licenseId) {
+      const mediaGallery: MediaObject[] = [];
+      const basicProfile = {
+        name: content.name ?? "",
+        url: content.url ?? "",
+      };
+
+      const mediaGalleryCid = await ipfs.dag.put(mediaGallery, {
+        storeCodec: "dag-cbor",
       });
 
-      const model = new DataModel({
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        ceramic: ceramic as any,
-        aliases: GeoWebModel,
+      const basicProfileCid = await ipfs.dag.put(basicProfile, {
+        storeCodec: "dag-cbor",
       });
 
-      const _assetContentManager = new AssetContentManager(
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        ceramic as any,
-        model,
-        ceramic.did?.capability.p.iss ?? "",
-        assetId
-      );
-
-      const _basicProfileStreamManager = new BasicProfileStreamManager(
-        _assetContentManager
-      );
-      await _basicProfileStreamManager.createOrUpdateStream(content);
-
-      setSelectedParcelId(`0x${new BN(parcelId.toString()).toString(16)}`);
-    } else if (basicProfileStreamManager) {
-      // Use existing BasicProfileStreamManager
-      await basicProfileStreamManager.createOrUpdateStream(content);
+      const parcelRoot = {
+        basicProfile: basicProfileCid,
+        mediaGallery: mediaGalleryCid,
+      };
+
+      const parcelRootCid = await ipfs.dag.put(parcelRoot, {
+        storeCodec: "dag-cbor",
+      });
+      const bytes = dagjson.encode(parcelRootCid);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const doc = await TileDocument.deterministic(ceramic as any, {
+        controllers: [ceramic.did?.parent ?? ""],
+        family: `geo-web-parcel`,
+        tags: [assetId.toString()],
+      });
+      await doc.update(json.decode(bytes));
+
+      setSelectedParcelId(`0x${new BN(licenseId.toString()).toString(16)}`);
+    } else {
+      try {
+        const ownerId = new AccountId(
+          AccountId.parse(ceramic.did?.parent.split("did:pkh:")[1] ?? "")
+        );
+        const web3Storage = new Web3Storage({
+          token: process.env.NEXT_PUBLIC_WEB3_STORAGE_TOKEN ?? "",
+          endpoint: new URL("https://api.web3.storage"),
+        });
+        const geoWebContent = new GeoWebContent({
```

Comment:
There only ever needs to be one instance of `GeoWebContent`. Perhaps this can be instantiated at the top level where `ceramic` and `ipfs` are instantiated? And then you would only need `parcelId` and `ownerId` where it's being called.